server/build_event_protocol/event_parser/event_parser_test.go

```diff
@@ -204,40 +204,40 @@ func TestFillInvocation(t *testing.T) {
-	assert.Equal(t, []string{"explicit"}, optionsParsed.ExplicitCmdLine)
+	assert.Equal(t, []string{"foo"}, invocation.Event[4].BuildEvent.GetOptionsParsed().CmdLine)
+	assert.Equal(t, []string{"explicit"}, invocation.Event[4].BuildEvent.GetOptionsParsed().ExplicitCmdLine)
 
-	assert.Equal(t, "uri", actionExecuted.Stdout.GetUri())
-	assert.Equal(t, "uri", actionExecuted.Stderr.GetUri())
-	assert.Equal(t, "uri", actionExecuted.PrimaryOutput.GetUri())
-	assert.Equal(t, 1, len(actionExecuted.ActionMetadataLogs))
-	assert.Equal(t, "uri", actionExecuted.ActionMetadataLogs[0].GetUri())
+	assert.Equal(t, "uri", invocation.Event[6].BuildEvent.GetAction().Stdout.GetUri())
+	assert.Equal(t, "uri", invocation.Event[6].BuildEvent.GetAction().Stderr.GetUri())
+	assert.Equal(t, "uri", invocation.Event[6].BuildEvent.GetAction().PrimaryOutput.GetUri())
+	assert.Equal(t, 1, len(invocation.Event[6].BuildEvent.GetAction().ActionMetadataLogs))
+	assert.Equal(t, "uri", invocation.Event[6].BuildEvent.GetAction().ActionMetadataLogs[0].GetUri())
 
-	assert.Equal(t, 1, len(namedSetOfFiles.Files))
-	assert.Equal(t, "uri", namedSetOfFiles.Files[0].GetUri())
+	assert.Equal(t, 1, len(invocation.Event[7].BuildEvent.GetNamedSetOfFiles().Files))
+	assert.Equal(t, "uri", invocation.Event[7].BuildEvent.GetNamedSetOfFiles().Files[0].GetUri())
 
-	assert.Equal(t, 1, len(targetComplete.ImportantOutput))
-	assert.Equal(t, "uri", targetComplete.ImportantOutput[0].GetUri())
+	assert.Equal(t, 1, len(invocation.Event[8].BuildEvent.GetCompleted().ImportantOutput))
+	assert.Equal(t, "uri", invocation.Event[8].BuildEvent.GetCompleted().ImportantOutput[0].GetUri())
 
-	assert.Equal(t, 1, len(testResult.TestActionOutput))
-	assert.Equal(t, "uri", testResult.TestActionOutput[0].GetUri())
+	assert.Equal(t, 1, len(invocation.Event[9].BuildEvent.GetTestResult().TestActionOutput))
+	assert.Equal(t, "uri", invocation.Event[9].BuildEvent.GetTestResult().TestActionOutput[0].GetUri())
 
-	assert.Equal(t, 1, len(testSummary.Passed))
-	assert.Equal(t, 1, len(testSummary.Failed))
-	assert.Equal(t, "uri", testSummary.Passed[0].GetUri())
-	assert.Equal(t, "uri", testSummary.Failed[0].GetUri())
+	assert.Equal(t, 1, len(invocation.Event[10].BuildEvent.GetTestSummary().Passed))
+	assert.Equal(t, 1, len(invocation.Event[10].BuildEvent.GetTestSummary().Failed))
+	assert.Equal(t, "uri", invocation.Event[10].BuildEvent.GetTestSummary().Passed[0].GetUri())
+	assert.Equal(t, "uri", invocation.Event[10].BuildEvent.GetTestSummary().Failed[0].GetUri())
 
 	assert.Equal(t, int64(1000), invocation.DurationUsec)
 
-	assert.Equal(t, "SHELL=/bin/bash", shellOption.OptionValue)
-	assert.Equal(t, "SECRET=<REDACTED>", secretOption.OptionValue)
+	assert.Equal(t, "SHELL=/bin/bash", invocation.StructuredCommandLine[0].Sections[0].GetOptionList().Option[0].OptionValue)
+	assert.Equal(t, "SECRET=<REDACTED>", invocation.StructuredCommandLine[0].Sections[0].GetOptionList().Option[1].OptionValue)
```

Comment:
I don't fully understand why these changes are necessary. I'm not totally opposed, but they definitely make the tests harder to read / modify since you now need to think about the event index.

Did these all fail or just the commandline one due to the change above?

Reply:
I agree strongly that it is harder to read; however, as it stands, `invocation` contains copies of all these protos (some few of which it alters), so the old code was comparing against the protos that were the input to invocation and confirming nothing about the actual work done by the parser to populate the invocation. This way we instead actually check against the values populated in `invocation` by the parser, which is presumably the point of the test.

Reply:
Even after implementing the change whereby we no longer make copies of the protos to store in `invocation`, we still need the test to ensure that the protos are being stored in `invocation` at all, so I think this change remains necessary even with the proposed signature change to `parseAndFilterCommandLine`